internal/service/remotecfg/remotecfg.go

```diff
@@ -401,6 +428,12 @@ func (s *Service) getAPIConfig() ([]byte, error) {
 		return nil, err
 	}
 	s.metrics.getConfigTime.Observe(time.Since(start).Seconds())
+	if gcr.Msg.Hash != "" {
+		s.remoteHash = gcr.Msg.Hash
+	}
+	if gcr.Msg.NotModified {
```

Comment:
super nit: this conditional returns so it can be done before the hash set on line 431 directly above